go-controller/pkg/ovn/controller/services/services_controller.go

```diff
@@ -234,8 +234,11 @@ func (c *Controller) syncServices(key string) error {
 	// These are the VIPs (ClusterIP:Port) that we have seen so far
 	// If the Service has updated the VIPs (has changed the Ports)
 	// and some were removed we have to delete those
-	// We need to create a NewString set to not mutate the service tracker VIPs
-	vipsTracked := sets.NewString().Union(c.serviceTracker.getService(name, namespace))
+	// We need to create a map to not mutate the service tracker VIPs
+	vipsTracked := make(map[string]string)
+	for k, v := range c.serviceTracker.getService(name, namespace) {
```

Comment:
I think vipsTracked can be a set.String now?
I checked in all the places we use, we only access the key.

This would result in changing the methods argument as well.

Reply:
will do